app/models/generic_event.rb

```diff
@@ -161,17 +161,10 @@ def self.relationship_attributes(attributes)
       @serializer ||=
         Class.new(GenericEventSerializer).tap do |serializer|
           relationship_attributes.each do |attribute_key, attribute_type|
-            # NB: Default to V2 relationship serializer and fallback to unversioned serializer. Never support v1
-            relation_serializer = "V2::#{attribute_type.to_s.singularize.camelize}Serializer"
-            relation_serializer = if const_defined?(relation_serializer)
-                                    relation_serializer
-                                  else
-                                    relation_serializer.gsub('V2::', '')
-                                  end
-
             named_relationship_key = attribute_key.to_s.sub('_id', '')
+
             serializer.set_type :events
-            serializer.has_one named_relationship_key, serializer: relation_serializer.constantize
+            serializer.has_one named_relationship_key, serializer: SerializerVersionChooser.call(attribute_type.to_s.singularize.camelize)
```

Comment:
It would be really nice to do this conversion inside the new chooser class if possible...

```suggestion
            new_serializer_class.has_one named_relationship_key, serializer: SerializerVersionChooser.call(attribute_type)
```

Reply:
That's fine for everything except the `AllocationsSerializer` which goes against the singular grain of `<model>Serializer` naming. We're unlikely to ever create an event about an Allocation record though? Since it's just a bag of individual moves?

Reply:
We can always make it simpler to handle that situation later, though.

Reply:
We have singular and plural versions of both move and allocation serializers now, but I would imagine that an event only relates to a singular resource so this shouldn't be a problem.